### PR TITLE
Add Members

### PR DESCRIPTION
**Pull request by zzybluebell**
AddMemberCommand extends from  abstract class AddCommand.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all members whose names contain any of "
             + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
             + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " alice bob charlie";
+            + "Example: " + COMMAND_WORD + " -m alice";
```
Comment (HolmesJJ) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:20]: I think extract the `-m` is better

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 2]
```diff
@@ -51,6 +52,7 @@
     public static final String ADDRESS_DESC_BOB = " " + PREFIX_ADDRESS + VALID_ADDRESS_BOB;
     public static final String TAG_DESC_FRIEND = " " + PREFIX_TAG + VALID_TAG_FRIEND;
     public static final String TAG_DESC_HUSBAND = " " + PREFIX_TAG + VALID_TAG_HUSBAND;
+    public static final String VALID_MEMBER = " " + PREFIX_MEMBER;
```
Comment (HolmesJJ) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:55]: Check the naming

File: src/test/java/seedu/address/testutil/EditMemberDescriptorBuilder.java [hunk 3]
```diff
@@ -36,10 +36,10 @@ public EditMemberDescriptorBuilder(Member member) {
         descriptor.setPhone(member.getPhone());
         descriptor.setEmail(member.getEmail());
         descriptor.setAddress(member.getAddress());
-        descriptor.setTags(member.getTags());
+        descriptor.setTags (member.getTags());
```
Comment (HolmesJJ) [inline src/test/java/seedu/address/testutil/EditMemberDescriptorBuilder.java:39]: Space

File: src/main/java/seedu/address/logic/commands/FindCommand.java (hunk 1 above)
Comment (zzybluebell) [reply src/main/java/seedu/address/logic/commands/FindCommand.java:20]: delete -m alreday

File: src/test/java/seedu/address/testutil/EditMemberDescriptorBuilder.java (hunk 3 above)
Comment (zzybluebell) [reply src/test/java/seedu/address/testutil/EditMemberDescriptorBuilder.java:39]: fixed

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java (hunk 2 above)
Comment (zzybluebell) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java]: Changed!

File: src/test/java/seedu/address/testutil/EditMemberDescriptorBuilder.java [hunk 4]
```diff
@@ -41,7 +41,7 @@ public EditMemberDescriptorBuilder(Member member) {
         descriptor.setTransactions(member.getTransactions());
     }
 
-    /**
+    /**s
```
Comment (HolmesJJ) [inline src/test/java/seedu/address/testutil/EditMemberDescriptorBuilder.java:44]: `s` should be removed

File: src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java [hunk 5]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (HolmesJJ) [inline src/test/java/seedu/address/logic/parser/CommandParserTestUtil.java:1]: remove empty line

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 6]
```diff
@@ -53,94 +54,95 @@ public void parse_allFieldsPresent_success() {
...
         // two invalid values, only first invalid value reported
-        assertParseFailure(parser, INVALID_NAME_DESC + PHONE_DESC_BOB + EMAIL_DESC_BOB
+        assertParseFailure(parser, MEMBER_DESC + INVALID_NAME_DESC + PHONE_DESC_BOB + EMAIL_DESC_BOB
                 + INVALID_ADDRESS_DESC, Name.MESSAGE_CONSTRAINTS);
 
         // non-empty preamble
         assertParseFailure(parser, PREAMBLE_NON_EMPTY + NAME_DESC_BOB + PHONE_DESC_BOB
-                + EMAIL_DESC_BOB + ADDRESS_DESC_BOB + TAG_DESC_HUSBAND + TAG_DESC_FRIEND,
-                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+                        + EMAIL_DESC_BOB + ADDRESS_DESC_BOB + TAG_DESC_HUSBAND + TAG_DESC_FRIEND,
```
Comment (HolmesJJ) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:145]: indentation not so correct

File: src/test/java/seedu/address/logic/parser/EzFoodieParserTest.java [hunk 7]
```diff
@@ -62,9 +62,11 @@
 
     @Test
     public void parseCommand_add() throws Exception {
+
```
Comment (HolmesJJ) [inline src/test/java/seedu/address/logic/parser/EzFoodieParserTest.java:65]: remove empty line

File: src/test/java/seedu/address/logic/parser/EzFoodieParserTest.java [hunk 8]
```diff
@@ -62,9 +62,11 @@
...
         Member member = new MemberBuilder(AMY).withCredit("0").withTransactions().build();
-        AddCommand command = (AddCommand) parser.parseCommand(MemberUtil.getAddCommand(member));
-        assertEquals(new AddCommand(member), command);
+        AddMemberCommand command = (AddMemberCommand) parser.parseCommand(MemberUtil.getAddCommand(member));
+        assertEquals(new AddMemberCommand(member), command);
+
```
Comment (HolmesJJ) [inline src/test/java/seedu/address/logic/parser/EzFoodieParserTest.java:69]: remove empty line

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 9]
```diff
@@ -53,94 +54,95 @@ public void parse_allFieldsPresent_success() {
...
         // missing address prefix
-        assertParseFailure(parser, NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY + VALID_ADDRESS_AMY,
+        assertParseFailure(parser, MEMBER_DESC + NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY + VALID_ADDRESS_AMY,
                 expectedMessage);
 
         // all prefixes missing
-        assertParseFailure(parser, VALID_NAME_AMY + VALID_PHONE_AMY + VALID_EMAIL_AMY + VALID_ADDRESS_AMY,
-                expectedMessage);
+        assertParseFailure(parser, MEMBER_DESC + VALID_NAME_AMY + VALID_PHONE_AMY + VALID_EMAIL_AMY
+                        + VALID_ADDRESS_AMY, expectedMessage);
```
Comment (HolmesJJ) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:114]: indentation not so correct

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 10]
```diff
@@ -53,94 +54,95 @@ public void parse_allFieldsPresent_success() {
...
+                + ADDRESS_DESC_BOB + ADDRESS_DESC_AMY + TAG_DESC_FRIEND, new AddMemberCommand(expectedMember));
 
         // multiple tags - all accepted
         Member expectedMemberMultipleTags = new MemberBuilder(AMY).withCredit("0")
                 .withTags(VALID_TAG_FRIEND, VALID_TAG_HUSBAND).withTransactions().build();
-        assertParseSuccess(parser, NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY
-                + ADDRESS_DESC_AMY + TAG_DESC_HUSBAND + TAG_DESC_FRIEND, new AddCommand(expectedMemberMultipleTags));
+        assertParseSuccess(parser, MEMBER_DESC + NAME_DESC_AMY + PHONE_DESC_AMY + EMAIL_DESC_AMY + ADDRESS_DESC_AMY
+                + TAG_DESC_HUSBAND + TAG_DESC_FRIEND,
+                new AddMemberCommand(expectedMemberMultipleTags));
```
Comment (HolmesJJ) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:81]: Extra next line

File: src/main/java/seedu/address/logic/commands/AddMemberCommand.java [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.member.Member;
+
+public class AddMemberCommand extends AddCommand {
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a member to the ezFoodie. "
```
Comment (HolmesJJ) [inline src/main/java/seedu/address/logic/commands/AddMemberCommand.java:17]: Extra space at the end

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 12]
```diff
@@ -44,9 +44,11 @@
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the member identified "
```
Comment (HolmesJJ) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:44]: I think the editcommand dont need to update

File: src/main/java/seedu/address/logic/commands/FindCommand.java [hunk 13]
```diff
@@ -27,6 +27,7 @@
```
Comment (HolmesJJ) [inline src/main/java/seedu/address/logic/commands/FindCommand.java:27]: I think the findcommand dont need to update

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 14]
```diff
@@ -1,67 +1,15 @@
...
-        requireNonNull(model);
-
-        if (model.hasMember(toAdd)) {
-            throw new CommandException(MESSAGE_DUPLICATE_MEMBER);
-        }
-
-        model.addMember(toAdd);
-        return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
-    }
+    public abstract CommandResult execute(Model model) throws CommandException;
```
Comment (HolmesJJ) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:13]: `@override` is missing?

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 15]
```diff
@@ -14,12 +14,12 @@
     public static final Prefix PREFIX_EMAIL = new Prefix("-e");
```
Comment (HolmesJJ) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:14]: I think the CliSyntax dont need to update

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 16]
```diff
@@ -46,6 +47,7 @@
     public static final String VALID_TRANSACTION_200 = "200.00";
     public static final String VALID_TRANSACTION_300 = "300.00";
 
+    public static final String MEMBER_DESC = " " + PREFIX_MEMBER;
```
Comment (HolmesJJ) [inline src/test/java/seedu/address/logic/commands/CommandTestUtil.java:50]: `MEMBER_DESC_AMY`

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java [hunk 17]
```diff
@@ -10,6 +10,7 @@
 import static seedu.address.logic.commands.CommandTestUtil.INVALID_NAME_DESC;
 import static seedu.address.logic.commands.CommandTestUtil.INVALID_PHONE_DESC;
 import static seedu.address.logic.commands.CommandTestUtil.INVALID_TAG_DESC;
+import static seedu.address.logic.commands.CommandTestUtil.MEMBER_DESC;
```
Comment (HolmesJJ) [inline src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:13]: `MEMBER_DESC_AMY`

File: src/test/java/seedu/address/testutil/EditMemberDescriptorBuilder.java (hunk 4 above)
Comment (zzybluebell) [reply src/test/java/seedu/address/testutil/EditMemberDescriptorBuilder.java]: removed "s"

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java (hunk 9 above)
Comment (zzybluebell) [reply src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: fixed

File: src/test/java/seedu/address/logic/parser/AddCommandParserTest.java (hunk 10 above)
Comment (zzybluebell) [reply src/test/java/seedu/address/logic/parser/AddCommandParserTest.java]: deleted

File: src/main/java/seedu/address/logic/commands/FindCommand.java (hunk 13 above)
Comment (zzybluebell) [reply src/main/java/seedu/address/logic/commands/FindCommand.java]: have not updated, delete the empty line

File: src/main/java/seedu/address/logic/parser/CliSyntax.java (hunk 15 above)
Comment (zzybluebell) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java]: deleted the empty line

**Pull request merged**